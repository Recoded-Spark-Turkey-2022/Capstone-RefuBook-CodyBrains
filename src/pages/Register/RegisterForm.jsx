import { yupResolver } from '@hookform/resolvers/yup';
import React from 'react';
import { useForm } from 'react-hook-form';
import { MdDriveFolderUpload } from 'react-icons/md';
import validationSchema from '../../utils/validation';
import useFirebaseFileUpload from '../../hooks/useFileUpload';

const RegisterForm = ({ onSubmit }) => {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm({
    resolver: yupResolver(validationSchema),
    mode: 'onBlur',
  });

  const { file, handleFileChange, percentage } = useFirebaseFileUpload;
  return (
    <form
      className="flex flex-col pt-3 md:pt-8  "
      onSubmit={handleSubmit(onSubmit)}
    >
      <div className="grid grid-cols-2 items-center justify-center gap-2">
        <div className="text-center">
          <label htmlFor="displayName">
            Username
            <input
              type="text"
              id="displayName"
              placeholder="Enter username"
              className="relative block w-full appearance-none rounded-none rounded-t-md border border-gray-300 px-3 py-2 focus:outline-none focus:border-refubookBlue mb-3 "
              {...register('displayName')}
            />
          </label>
          <p>{errors.displayName?.message}</p>
        </div>
        <div className="text-center">
          <label htmlFor="email">
            Email address
            <input
              type="email"
              id="email"
              name="email"
              placeholder="Enter your email"
              className="relative block w-full appearance-none rounded-none rounded-t-md border border-gray-300 px-3 py-2 focus:outline-none focus:border-refubookBlue mb-3 "
              {...register('email')}
            />
          </label>
          <p>{errors.email?.message}</p>
        </div>
        <div className="text-center">
          <label htmlFor="password">
            Password
            <input
              type="password"
              id="password"
              name="password"
              placeholder="Password"
              className="relative block w-full appearance-none rounded-none rounded-t-md border border-gray-300 px-3 py-2 focus:outline-none focus:border-refubookBlue mb-3 "
              {...register('password')}
            />
          </label>
          <p>{errors.password?.message}</p>
        </div>
        <div className="text-center">
          <label htmlFor="password2">
            Confirm Password
            <input
              type="password"
              id="password2"
              name="password2"
              placeholder="Confirm Password"
              className="relative block w-full appearance-none rounded-none rounded-t-md border border-gray-300 px-3 py-2 focus:outline-none focus:border-refubookBlue mb-3 "
              {...register('password2')}
            />
          </label>
          <p>{errors.password2?.message}</p>
        </div>
        <div className="text-center flex items-center justify-center">
          <img
            src={
              file
                ? URL.createObjectURL(file)
                : 'https://icon-library.com/images/no-image-icon/no-image-icon-0.jpg'
            }
            alt=""
            className="w-32 h-32 rounded-full"
          />
        </div>
        <div className="flex flex-row-reverse w-full items-center justify-center gap-4">
          <label
            htmlFor="photoURL"
            className="flex flex-col-reverse items-center gap-2"
          >
            Please upload a profile picture
            <MdDriveFolderUpload className="text-5xl text-refubookBlue" />
            <input
              type="file"
              id="photoURL"
              name="photoURL"
              className="hidden"
              {...register('photoURL')}
              onChange={handleFileChange}
            />
            {percentage > 0 && (
              <div className="w-full h-2 bg-gray-300 rounded-full">
                <div
                  className="h-full bg-refubookBlue rounded-full"
                  style={{ width: `${percentage}%` }}
                />
              </div>
            )}
          </label>
        </div>
        <p>{errors.photoURL?.message}</p>
      </div>
      <button
        type="submit"
        className=" className='group relative flex w-full justify-center rounded-md border border-transparent bg-refubookBlue py-2 px-4 text-sm font-medium mt-8 text-white hover:bg-refubookLightBlue focus:outline-none focus:ring-2 focus:ring-[#331416] focus:ring-offset-2'"
      >
        Register
      </button>
    </form>
  );
};

export default RegisterForm;
