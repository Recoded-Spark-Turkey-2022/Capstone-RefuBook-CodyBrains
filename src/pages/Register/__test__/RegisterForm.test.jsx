import renderer from 'react-test-renderer';
import RegisterForm from '../RegisterForm';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: () => <div>Link</div>,
}));

describe('RegisterForm', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<RegisterForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
