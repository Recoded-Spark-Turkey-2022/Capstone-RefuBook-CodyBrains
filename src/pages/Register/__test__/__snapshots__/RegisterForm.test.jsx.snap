// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm should render correctly 1`] = `
<form
  className="flex flex-col pt-3 md:pt-8  "
  onSubmit={[Function]}
>
  <div
    className="grid grid-cols-2 items-center justify-center gap-2"
  >
    <div
      className="text-center"
    >
      <label
        htmlFor="displayName"
      >
        Username
        <input
          className="relative block w-full appearance-none rounded-none rounded-t-md border border-gray-300 px-3 py-2 focus:outline-none focus:border-refubookBlue mb-3 "
          id="displayName"
          name="displayName"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter username"
          type="text"
        />
      </label>
      <p />
    </div>
    <div
      className="text-center"
    >
      <label
        htmlFor="email"
      >
        Email address
        <input
          className="relative block w-full appearance-none rounded-none rounded-t-md border border-gray-300 px-3 py-2 focus:outline-none focus:border-refubookBlue mb-3 "
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter your email"
          type="email"
        />
      </label>
      <p />
    </div>
    <div
      className="text-center"
    >
      <label
        htmlFor="password"
      >
        Password
        <input
          className="relative block w-full appearance-none rounded-none rounded-t-md border border-gray-300 px-3 py-2 focus:outline-none focus:border-refubookBlue mb-3 "
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </label>
      <p />
    </div>
    <div
      className="text-center"
    >
      <label
        htmlFor="password2"
      >
        Confirm Password
        <input
          className="relative block w-full appearance-none rounded-none rounded-t-md border border-gray-300 px-3 py-2 focus:outline-none focus:border-refubookBlue mb-3 "
          id="password2"
          name="password2"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Confirm Password"
          type="password"
        />
      </label>
      <p />
    </div>
    <div
      className="text-center flex items-center justify-center"
    >
      <img
        alt=""
        className="w-32 h-32 rounded-full"
        src="https://icon-library.com/images/no-image-icon/no-image-icon-0.jpg"
      />
    </div>
    <div
      className="flex flex-row-reverse w-full items-center justify-center gap-4"
    >
      <label
        className="flex flex-col-reverse items-center gap-2"
        htmlFor="photoURL"
      >
        Please upload a profile picture
        <svg
          className="text-5xl text-refubookBlue"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 6h-8l-2-2H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm0 12H4V8h16v10zM8 13.01l1.41 1.41L11 12.84V17h2v-4.16l1.59 1.59L16 13.01 12.01 9 8 13.01z"
          />
        </svg>
        <input
          className="hidden"
          id="photoURL"
          name="photoURL"
          onBlur={[Function]}
          type="file"
        />
      </label>
    </div>
    <p />
  </div>
  <button
    className=" className='group relative flex w-full justify-center rounded-md border border-transparent bg-refubookBlue py-2 px-4 text-sm font-medium mt-8 text-white hover:bg-refubookLightBlue focus:outline-none focus:ring-2 focus:ring-[#331416] focus:ring-offset-2'"
    type="submit"
  >
    Register
  </button>
</form>
`;
